addons/pos_gift_card/static/src/js/GiftCardPopup.js

```diff
@@ -42,12 +45,21 @@ odoo.define("pos_gift_card.GiftCardPopup", function (require) {
         this.env.pos.db.product_by_id[
           this.env.pos.config.gift_card_product_id[0]
         ];
-      this.env.pos.get_order().add_product(gift, {
-        price: this.state.amountToSet,
-        quantity: 1,
-        merge: false,
-        generated_gift_card_ids: giftCard ? giftCard.id : false,
-      });
+      
+      if (!giftCard.buy_pos_order_line_id) {
+        this.env.pos.get_order().add_product(gift, {
+          price: this.state.amountToSet,
+          quantity: 1,
+          merge: false,
+          generated_gift_card_ids: giftCard ? giftCard.id : false,
+        });
+      }
+      else {
+        Gui.showPopup('ErrorPopup', {
+          'title': _t('This gift card has already been sold'),
+          'body': _t('You cannot sell a gift card that has already been sold')
```

Comment:
If you add translatable terms, you should add the corresponding entries in the pot file.